library/core/src/ptr/const_ptr.rs

```diff
@@ -409,6 +409,20 @@ impl<T: ?Sized> *const T {
     #[rustc_const_stable(feature = "const_ptr_offset", since = "1.61.0")]
     #[inline(always)]
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+    #[requires(
+        // Precondition 1: the computed offset `count * size_of::<T>()` does not overflow `isize`
+        count.checked_mul(core::mem::size_of::<T>() as isize).is_some() &&
+        // Precondition 2: adding the computed offset to `self` does not cause overflow
+        (self as isize).checked_add((count * core::mem::size_of::<T>() as isize)).is_some() &&
```

Comment:
There is an implicit condition here. What happens if `core::mem::size_of::<T>()` represents a *very* larger type in a 64-bit system? Wouldn't the conversions to `isize` resolve this to checking if the wrong negative number fits the specification? I'm a bit curious on if you gave that a thought. For the types that you instantiated in the harnesses, this will not be a problem at all. This is not a blocker.

Reply:
Based on the information in the Rust documentation [here](https://doc.rust-lang.org/stable/reference/types/numeric.html#machine-dependent-integer-types), the size of `T` cannot exceed `isize::MAX` on any platform. As a result, the conversion from `usize` to `isize` will not cause an overflow, ensuring that `core::mem::size_of::<T>() as isize` will always be non-negative.